tests/shim_test.go

```diff
@@ -188,6 +188,41 @@ func TestShimExitWithIncorrectBase64Token(t *testing.T) {
 	rig.Stop()
 }
 
+func TestShimExitWithNonBase64Token(t *testing.T) {
+	rig := newTestRig(t)
+	rig.Start(invalidToken)
+	assert.NotNil(t, rig.proxy)
+
+	//Wait for shim to connect to exit
+	cmd := rig.shimCommand
+	assert.NotNil(rig.t, cmd.Process)
+	err := cmd.Wait()
+	assert.NotNil(rig.t, err)
+
+	timedout := false
+	timer := make(chan bool, 1)
```

Comment:
Why are you buffering the channel ? You are already using a go routine, you should be fine without buffering.